packages/models/src/delegation/delegationEvents.ts

```diff
@@ -0,0 +1,60 @@
+import { z } from "zod";
+import { match } from "ts-pattern";
+
+import {
+  DelegationSubmittedV2,
+  DelegationApprovedV2,
+  DelegationRejectedV2,
+  DelegationRevokedV2,
+} from "../gen/v2/delegation/events.js";
+import { protobufDecoder } from "../protobuf/protobuf.js";
+import { EventEnvelope } from "../events/events.js";
+
+export const DelegationEventV2 = z.discriminatedUnion("type", [
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationSubmittedV2"),
+    data: protobufDecoder(DelegationSubmittedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationApprovedV2"),
+    data: protobufDecoder(DelegationApprovedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationRejectedV2"),
+    data: protobufDecoder(DelegationRejectedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationRevokedV2"),
+    data: protobufDecoder(DelegationRevokedV2),
+  }),
+]);
+
+export type DelegationEventV2 = z.infer<typeof DelegationEventV2>;
+
+export function delegationEventToBinaryDataV1(
```

Comment:
```suggestion
export function delegationEventToBinaryDataV2(
```

Reply:
Good catch for the possible future improvements! 👍🏻 
This commit [2ecf569](https://github.com/pagopa/interop-be-monorepo/pull/1053/commits/2ecf569f5ac8ee8db784b50ae3e1a1894b53a417) add discriminant type and some ranaming